uniswap_swap_detection/src/agent.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+} from "forta-agent";
+
+import { SWAP_EVENT } from "./constant";
+import { poolABI } from "./abi";
+
+import { uniswapPoolAddress } from "./utils";
+
+interface HandleArgs {
+  PROVIDER: ethers.providers.JsonRpcProvider;
+}
+const PROVIDER = getEthersProvider();
+const block = 20;
+export const provideHandleTransaction =
+  (PROVIDER: ethers.providers.JsonRpcProvider, block: Number) =>
```

Comment:
I don't understand what is the need of the `block: Number` argument here? `TransactionEvent` has a property of `blockNumber` that you can use. Source: https://docs.forta.network/en/latest/sdk/#transactionevent

Reply:
Also, I don't think it's necessary to name the argument `PROVIDER` in all caps. That's pretty much reserved for constants. So as an argument name, `provider` would be enough.